engine/source/elements/shell/coquez/czforc3.F

```diff
@@ -697,10 +700,15 @@ SUBROUTINE CZFORC3(
 C-------------------------
 c     Thermique des coques 
 C--------------------------
-        IF (JTHE > 0) THEN  
+        IF (JTHE /= 0) THEN  
+          IF (MAT_ELEM%MAT_PARAM(IMAT)%HEAT_FLAG == 1) THEN
+            HEAT => FHEAT(1:NEL)
+          ELSE
+            HEAT => DIE(1:NEL)
+          END IF
           CALL THERMC(JFT    ,JLT     ,PM      ,MAT  ,THK0 ,IXC  ,
```

Comment:
We could duplicate the call to THERMC instead of using pointers here. See the compiler warnings:


Warning: Creating array temporary at (1) [-Warray-temporaries]